src/runtime/cache.ts

```diff
@@ -78,11 +80,13 @@ export function defineCachedFunction<T = any>(
 
     const _resolve = async () => {
       if (!pending[key]) {
-        // Remove cached entry to prevent using expired cache on concurrent requests
-        entry.value = undefined;
-        entry.integrity = undefined;
-        entry.mtime = undefined;
-        entry.expires = undefined;
+        if (entry.value && !opts.allowStale) {
```

Comment:
Does it makes sense that we directly use `!opts.swr` here?

Reply:
You mean instead of `entry.value && !opts.allowStale`?

Reply:
Yes. I suppose it should be expected behavior of SWR to be nonblocking after `maxAge`. (To be precise, also before `staleMaxAge` but it is not implemented yet in handler)

Reply:
You are definitely not wrong. We use cache options for both Browser<>Server And Server<>Handler caching strategy. And these utils handle Server<>Handler SWR caching strategy _similar_ to browsers (regardless of HTTP client which could be another API or browser)

----

SWR Cache Timeline: `0....maxAge` (A) .... `maxAge...staleMaxAge` (B) ... after maxAge/staleMaxAge (C)

Stage A: Fresh Cache. Use value without revalidate
Stage B: Stale Cache. Respond and revalidate in background
Stage C: Expired Cache. Block and revalidate

Since we didn't implement `staleMaxAge` yet, it is either B or C. New `allowStale` flag is identical to `staleMaxAge: Infinity`.

How about supporting `staleMaxAge: -1` as a way to support without new flag?

Reply:
Alright, yes I do see what you mean and it would make sense to use the existing `staleMaxAge` flag to avoid multiplying the options; i'll update the PR